.esh/commands/reshape_i18n_keys.py

```
@@ -0,0 +1,433 @@
+			return
+		parent = parent[part]
+	if isinstance(parent, dict):
+		parent.pop(parts[-1], None)
+
+
+def set_at(obj, path: str, value) -> None:
+	parts = path.split(".")
+	cur = obj
+	for i, part in enumerate(parts[:-1]):
+		if not isinstance(cur, dict):
+			raise TypeError(f"Cannot set {path!r}: {'.'.join(parts[:i])!r} is not a dict")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)